products/accesscontextmanager/api.yaml

```diff
@@ -66,3 +66,200 @@ objects:
         description: |
           Time the AccessPolicy was updated in UTC.
         output: true
+  - !ruby/object:Api::Resource
+    name: 'AccessLevel'
+    # This is a weird API, so we need to use a few fields to map the methods to
+    # the right URL.
+    # create_url is the Create URL
+    # base_url is the Get and Delete and Patch URL. It is empty on purpose.
+    # List won't work yet. It should share a URL with Create.
+    create_url: |
+      {{parent}}/accessLevels
+    base_url: |
+
+    self_link: |
+      {{name}}
+    update_verb: :PATCH
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Access Policy Quickstart': 'https://cloud.google.com/access-context-manager/docs/quickstart'
+      api: 'https://cloud.google.com/access-context-manager/docs/reference/rest/v1beta/accessPolicies.accessLevels'
+    description: |
+      An AccessLevel is a label that can be applied to requests to GCP services,
+      along with a list of requirements necessary for the label to be applied.
+<%= indent(compile_file({}, 'templates/global_async.yaml.erb'), 4) %>
+    parameters:
+      # Parent is a path parameter that _cannot_ be read or sent in the request at all.
+      # This must be done at the provider level.
+      - !ruby/object:Api::Type::String
+        name: parent
+        input: true
+        required: true
+        description: |
+          The AccessPolicy this AccessLevel lives in.
+          Format: accessPolicies/{policy_id}
+      - !ruby/object:Api::Type::String
+        name: name
```

Comment:
So I looked around and the redis API is another similar one to this where the name is the long-style and required. That one I handled by pretending that it's actually the short form and using a custom expander/flattener. The big advantage of that approach is consistency across resources, the disadvantage is that interpolation on it won't work directly. Do you have an opinion on whether one is better than the other for this resource?

Reply:
I'm not sure what my feelings are on how to handle newer style `relative_uri` shaped `name` fields, and I figured a fully qualified name would be the most future proof until https://github.com/GoogleCloudPlatform/magic-modules/issues/937 decides on how we handle these formats consistently across products.